client/js/components/home/teamCreationForm.js

```diff
@@ -79,28 +81,48 @@ var TeamCreationForm = React.createClass({
         this.setState({tags: _.uniq(tags), creatingTag: false})
       },
       error: function (xhr, status, err) {
+        console.log(xhr.status + ' ' + xhr.statusText)
+        console.log(xhr.responseText)
         var message = 'Tag creation failed due to unknown reason. Try again later.'
+        if (xhr.status === HttpStatus.BAD_REQUEST) {
+          if (xhr.responseJSON && xhr.responseJSON.name) {
+            message = 'Invalid tag name: ' + xhr.responseJSON.name
+          } else {
+            message = 'Tag creation failed: ' + xhr.responseText
+          }
+        }
         this.showError(message)
       }
     })
   },
 
   // FIXME Temporary until react-select fixes allowCreate={true}
```

Comment:
@mvillis I'm not sure about this change... it seems to violate the principle of least surprise.

"But that's not what I typed!"